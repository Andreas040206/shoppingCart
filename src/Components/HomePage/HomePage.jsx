import homePageStyles from "./HomePage.module.css";

import frontImgSrc from "../../assets/frontClothingImg.jpg";
import logoPngSrc from "../../assets/Logo-removebg.png";

export const HomePage = () => {
  return (
    <div className={homePageStyles.page}>
      <div className={homePageStyles.imgCon}>
        <img className={homePageStyles.logo} src={logoPngSrc} />
        <img className={homePageStyles.coverImg} src={frontImgSrc} />
      </div>
      <div className={homePageStyles.textCon}>
        <span className={homePageStyles.header}>Fashion since 1923</span>
        <span className={homePageStyles.text}>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent
          rutrum scelerisque rutrum. Donec auctor magna sit amet consequat
          pretium. Ut mollis sollicitudin ex eget consectetur. Duis luctus
          elementum ex, condimentum aliquam nisl ultrices vel. Nullam vel
          suscipit justo. Ut vel dignissim risus. Nam tristique laoreet massa
          auctor lacinia. Sed sit amet porttitor est. Suspendisse rhoncus velit
          id mattis malesuada.
        </span>
      </div>
    </div>
  );
};
