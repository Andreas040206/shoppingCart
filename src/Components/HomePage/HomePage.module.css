.page {
  display: flex;
  flex-direction: column;
}

.imgCon {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  position: relative;
}

.coverImg {
  width: 100%;
  height: min(600px, 100vh);
  object-fit: cover;
  object-position: 0px -15vw;
}

.logo {
  position: absolute;
  filter: brightness(0%);
  height: 160px;
  animation: logoAnimation 7s infinite;
}

.textCon {
  display: flex;
  flex-direction: column;
  padding: 100px max(5vw, calc(50vw - 600px));
  gap: 20px;
}

.header {
  font-size: 40px;
  color: rgb(68, 0, 0);
}

.text {
  font-size: 20px;
  color: rgb(68, 0, 0);
}

@keyframes logoAnimation {
  0% {
    filter: brightness(0%);
  }
  10% {
    filter: brightness(0%);
  }
  30% {
    filter: brightness(100%);
  }
  70% {
    filter: brightness(100%);
  }
  90% {
    filter: brightness(0%);
  }
  100% {
    filter: brightness(0%);
  }
}
