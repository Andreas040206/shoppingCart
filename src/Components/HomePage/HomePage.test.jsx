import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";

import { HomePage } from "./HomePage.jsx";

// Their is no logic or any interactive parts in the homepage therfor i only test the snapshot
describe("Homepage test", () => {
  it("Renders the content correctly", () => {
    const { container } = render(<HomePage />);
    expect(container).toMatchSnapshot();
  });
});
