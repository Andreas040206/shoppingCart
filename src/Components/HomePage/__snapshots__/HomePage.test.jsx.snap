// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage test > Renders the content correctly 1`] = `
<div>
  <div
    class="_page_99d6b3"
  >
    <div
      class="_imgCon_99d6b3"
    >
      <img
        class="_logo_99d6b3"
        src="/src/assets/Logo-removebg.png"
      />
      <img
        class="_coverImg_99d6b3"
        src="/src/assets/frontClothingImg.jpg"
      />
    </div>
    <div
      class="_textCon_99d6b3"
    >
      <span
        class="_header_99d6b3"
      >
        Fashion since 1923
      </span>
      <span
        class="_text_99d6b3"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent rutrum scelerisque rutrum. Donec auctor magna sit amet consequat pretium. Ut mollis sollicitudin ex eget consectetur. Duis luctus elementum ex, condimentum aliquam nisl ultrices vel. Nullam vel suscipit justo. Ut vel dignissim risus. Nam tristique laoreet massa auctor lacinia. Sed sit amet porttitor est. Suspendisse rhoncus velit id mattis malesuada.
      </span>
    </div>
  </div>
</div>
`;
